src/server/pfs/server/driver.go

```diff
@@ -1768,152 +1432,84 @@ func (d *driver) createBranch(txnCtx *txncontext.TransactionContext, branch *pfs
 	if len(provenance) > 0 && trigger != nil {
 		return errors.New("a branch cannot have both provenance and a trigger")
 	}
-
 	var err error
 	if err := d.env.AuthServer.CheckRepoIsAuthorizedInTransaction(txnCtx, branch.Repo, auth.Permission_REPO_CREATE_BRANCH); err != nil {
 		return errors.EnsureStack(err)
 	}
 	// Validate request
-	if err := ancestry.ValidateName(branch.Name); err != nil {
+	allBranches := make([]*pfs.Branch, 0)
+	allBranches = append(allBranches, provenance...)
+	allBranches = append(allBranches, branch)
+	for _, b := range allBranches {
+		if err := ancestry.ValidateName(b.Name); err != nil {
+			return err
+		}
+	}
+	// Create any of the provenance branches that don't exist yet, and give them an empty commit
+	if err := d.fillNewBranches(txnCtx, branch, provenance); err != nil {
 		return err
 	}
-
-	// Retrieve (and create, if necessary) the current version of this branch
+	// if the user passed a commit to point this branch at, resolve it
+	var ci *pfs.CommitInfo
+	if commit != nil {
+		ci, err = d.resolveCommit(txnCtx.SqlTx, commit)
+		if err != nil {
+			return errors.Wrapf(err, "unable to inspect %s", commit)
+		}
+		commit = ci.Commit
+	}
+	// Retrieve the current version of this branch and set it's head if specified
```

Comment:
‘its’

As an aside, I kind of think that comments should either be uncapitalised sentence fragments or properly-capitalised sentences ending in periods, but … it doesn’t really matter.